### DISCUSSION TITLE
Adding seurat metadata to ArchR Project

### DISCUSSION THREAD

**Discussion by ebulger3** [GreenleafLab/ArchR#1464] [Questions / Documentation]
Hi Ryan, 
I have 10x multome dataset and I am trying to import cluster labels from Seurat into my ArchR project ("archr_proj"). I currently have a dataframe with my cell barcodes and corresponding Seurat clusters, and have been trying to adapt the for-loop from #788.  A few lines from my dataframe ("seRNA_df") are below:

> head(seRNA_df)
                                                        seurat_clust
eb01#AAACAGCCAATAAGCA-1 Primitive Streak-1
eb01#AAACAGCCAATTGAGA-1              PGCLC
eb01#AAACAGCCAGCTAACC-1           Mesoderm
eb01#AAACAGCCATGCTCCC-1 Primitive Streak-2
eb01#AAACATGCAAGCTTAT-1          Undefined
eb01#AAACATGCAGAAACGT-1           Epiblast

This is the code I adapted from #788:

```
 archr_proj@cellColData[,colnames(seRNA_df)[1]] <- NA

  for(x in 1:nrow(seRNA_df)) {
    cellsToChange <- getCellNames(ArchRProj = archr_proj)[which(as.character(archr_proj@cellColData$cellNames) == rownames(seRNA_df)[x])] 
    archr_proj@cellColData[cellsToChange,colnames(seRNA_df)[1]] <-seRNA_df[x,1]
  }
```

This gives me "NA" for every cell in the "seurat_clust" column of the ArchR project, despite having overlapping barcodes. I suspect the issue is a small nuance of number juggling, but I am having a really hard time figuring out where the problem is. I'd appreciate any suggestions. 

Thanks!

**Comment by rcorces** [comment]
This isnt really an ArchR question. What your question boils down to is how do I transfer information from one data frame to another based on matched values across the two data frames. The key is obviously making sure that the matched values (cellNames in both dataframes) actually match.
This should be decipherable with google or other support forums.

**Reply by ebulger3** [reply]
From everything I can tell the matched values do match, but thank you for confirming it's not a general issue of how I am extracting information from the archR project. I will keep looking in other support forums, thanks.
